include/fiction/algorithms/simulation/sidb/displacement_robustness_domain.hpp

```diff
@@ -0,0 +1,670 @@
+                        {
+                            new_pos_nw.x -= static_cast<decltype(new_pos_nw.x)>(params.displacement_variations.first);
+                            new_pos_se.x += static_cast<decltype(new_pos_se.x)>(params.displacement_variations.first);
+                        }
+
+                        // treating the y-coordinate
+                        if (params.dimer_policy == dimer_displacement_policy::STAY_ON_ORIGINAL_DIMER &&
+                            params.displacement_variations.second > 0)
+                        {
+                            auto new_pos_se_siqad = fiction::siqad::to_siqad_coord(c);
+                            auto new_pos_nw_siqad = fiction::siqad::to_siqad_coord(c);
+
+                            new_pos_se_siqad.z = 0;
+                            new_pos_nw_siqad.z = 1;
+
+                            new_pos_nw = fiction::siqad::to_fiction_coord<coordinate<Lyt>>(new_pos_nw_siqad);
+                            new_pos_se = fiction::siqad::to_fiction_coord<coordinate<Lyt>>(new_pos_se_siqad);
+                        }
+
+                        else
+                        {
+                            if constexpr (has_offset_ucoord_v<Lyt>)
+                            {
+                                if (new_pos_nw.y <
+                                    static_cast<decltype(new_pos_nw.y)>(params.displacement_variations.second))
+                                {
+                                    new_pos_nw.y = 0;
+                                }
+                                else
+                                {
+                                    new_pos_nw.y -=
+                                        static_cast<decltype(new_pos_nw.y)>(params.displacement_variations.second);
+                                }
+                                new_pos_se.y +=
+                                    static_cast<decltype(new_pos_se.y)>(params.displacement_variations.second);
+                            }
+                            else
+                            {
+                                new_pos_nw.y -=
+                                    static_cast<decltype(new_pos_nw.y)>(params.displacement_variations.second);
+                                new_pos_se.y +=
+                                    static_cast<decltype(new_pos_se.y)>(params.displacement_variations.second);
+                            }
+                        }
+                    }
+
+                    const auto all_coord = all_coordinates_in_spanned_area<cell<Lyt>>(new_pos_se, new_pos_nw);
+                    all_possible_sidb_misplacements.push_back(all_coord);
+                }
+            });
+
+        all_possible_sidb_displacements = all_possible_sidb_misplacements;
+    }
+    /**
+     * This is a helper function, which recursively generates combinations of SiDB displacements for all SiDBs
+     * based on the provided vector of displacement vectors.
```

Comment:
What combinations? Can we make this more explicit? Can we use the `combinations.h` library for this use case?

Reply:
I'm not quite sure as the indices here are not unique, e.g. [0,0,1,10,3,0,1,1] can be a combination and I don't know, or I haven't found, how or if this can be done with ``combinations.h``.